test/run.d

```diff
@@ -0,0 +1,308 @@
+        auto exe = env["EXE"] = ".exe";
+        env["OBJ"] = ".obj";
+        env["DSEP"] = `\\`;
+        env["SEP"] = `\`;
+        auto druntimePath = environment.get("DRUNTIME_PATH", `..\..\druntime`);
+        auto phobosPath = environment.get("PHOBOS_PATH", `..\..\phobos`);
+        env["DFLAGS"] = `-I%s\import -I%s`.format(druntimePath, phobosPath);
+        env["LIB"] = phobosPath;
+
+        // auto-tester might run the testsuite with a different $(MODEL) than DMD
+        // has been compiled with. Hence we manually check which binary exists.
+        // For windows the $(OS) during build is: `windows`
+        int dmdModel = "../generated/windows/%s/64/dmd%s".format(build, exe).exists ? 64 : 32;
+        env.getDefault("MODEL", dmdModel.text);
+        env["DMD"] = "../generated/windows/%s/%d/dmd%s".format(build, dmdModel, exe);
+    }
+    else
+    {
+        env.getDefault("ARGS", "-inline -release -g -O -fPIC");
+        env["EXE"] = "";
+        env["OBJ"] = ".o";
+        env["DSEP"] = "/";
+        env["SEP"] = "/";
+        auto druntimePath = environment.get("DRUNTIME_PATH", `../../druntime`);
+        auto phobosPath = environment.get("PHOBOS_PATH", `../../phobos`);
+
+        // auto-tester might run the testsuite with a different $(MODEL) than DMD
+        // has been compiled with. Hence we manually check which binary exists.
+        int dmdModel = "../generated/%s/%s/64/dmd".format(os, build).exists ? 64 : 32;
+        env.getDefault("MODEL", dmdModel.text);
+        auto generatedPath = "generated/%s/%s/%s/dmd".format(os, build, dmdModel);
+        env["DMD"] = "../" ~ generatedPath;
+
+        // default to PIC on x86_64, use PIC=1/0 to en-/disable PIC.
+        // Note that shared libraries and C files are always compiled with PIC.
+        bool pic;
+        version(X86_64)
+            pic = true;
+        else version(X86)
+            pic = false;
+
+        env["PIC_FLAGS"]  = pic ? "-fPIC" : "";
+        env["DFLAGS"] = "-I%s/import -I%s".format(druntimePath, phobosPath)
+            ~ " -L-L%s/%s".format(phobosPath, generatedPath);
+        bool isShared = os.among("linux", "freebsd") >= 0;
+        if (isShared)
+            env["DFLAGS"] = env["DFLAGS"] ~ " -defaultlib=libphobos2.so -L-rpath=%s/%s".format(phobosPath, generatedPath);
+
+        env["REQUIRED_ARGS"] = environment.get("REQUIRED_ARGS") ~  env["PIC_FLAGS"];
+    }
+    return env;
+}
+
+/*
+Detects the host OS.
+
+Returns: a string from `{osx,linux,freebsd,openbsd,netbsd,dragonflybsd,solaris}`
+*/
+string detectOS()
+{
+    version(OSX)
+        return "osx";
+    else version(linux)
+        return "linux";
+    else version(FreeBSD)
+        return "freebsd";
+    else version(OpenBSD)
+        return "openbsd";
+    else version(NetBSD)
+        return "netbsd";
+    else version(DragonFlyBSD)
+        return "dragonflybsd";
+    else version(Solaris)
+        return "solaris";
+    else version(SunOS)
+        return "solaris";
+    else
```

Comment:
No windows!??!! :)

I went ahead and tested that this runs on windows and it looks like it does.  Just had to add the `version(Windows)` case here.

Reply:
(windows is on the top of this list now)